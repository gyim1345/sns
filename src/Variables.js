export const post = {
  like: [('guy@gmail.com', 'noone@gmail.com', 'gibong@gmail.com')],
  tag: [('boat', 'sky', 'landscape')],
  _id: '5e8ec52d9bb4913b5a844423',
  id: 1,
  title: 'posting with id 1',
  imageUrl: 'http://localhost:3000/static/images/1.jpg',
  userName: 'gibong@gmail.com'
};

export const posts = [
  {
    like: [('guy@gmail.com', 'noone@gmail.com', 'gibong@gmail.com')],
    tag: [('boat', 'sky', 'landscape')],
    _id: '5e8ec52d9bb4913b5a844423',
    id: 1,
    title: 'posting with id 1',
    imageUrl: 'http://localhost:3000/static/images/1.jpg',
    userName: 'gibong@gmail.com'
  }
];

export const currentUser = 'gibong@gmail.com';

export const currentUserName = 'gibong'

export const comments = [
  {
    like: [('gibong@gmail.com', 'guy@gmail.com', 'noone@gmail.com')],
    _id: '5e90119435a2d94f9e08e35a',
    id: 1,
    postLId: 1,
    title: 'title1',
    userName: 'gibong@gmail.com',
    replyToCommentId: null
  }
];

export const info = {
  userURL: 'http://localhost:3000/static/images/user1.png',
  name: 'gibong@gmail.com',
  postCount: 16,
  followerNumber: 3,
  nickName: '기봉이',
  introductory: '나는 기봉이'
};

export const sizeOfPicture = {
  width: '10px',
  height: '10px'
};
