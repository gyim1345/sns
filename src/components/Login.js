import React from "react";
import { useForm } from "react-hook-form";
import PropTypes from "prop-types";
import { Redirect } from "react-router-dom";
import userStorage from "../stores/userStore";
import { setLoginAPI } from "../apis/login"

const Login = ({
  setUserOfActivePage,
  setCurrentUser,
  setLoggedIn,
  toggleLogInStatus,
  loggedIn,
  currentUser
}) => {
  const { register, handleSubmit, watch, errors } = useForm();

  const checkIdIsRegistered = (data) => {
    return userStorage.userList.find(item => item.name === data.Id) === undefined
  }

  const checkPassword = (data) => {
    return userStorage.getUserPassword(data.Id) !== data.Password
  }

  const performLogIn = (data) => {
    return (
      setCurrentUser(data.Id),
      setUserOfActivePage(data.Id),
      toggleLogInStatus(),
      alert("logged in"),
      setLoggedIn(true)
    )
  }

  const onSubmit = async (data) => {
    console.log(data)
    try {
      const response = await setLoginAPI(data);
      console.log(response);
      alert(response.statusMessage)
      response.loginStatus
        && (setCurrentUser(data.Id),
          setUserOfActivePage(data.Id))
      // && toggleLogInStatus()
      setLoggedIn(response.loginStatus)

    } catch (e) {
      console.log(e)
    }
  };

  console.log(currentUser , loggedIn)

  // const onSubmit = data => {
  //   console.log(userStorage.userList.find(item => item.name === data.Id) === undefined)
  //     checkIdIsRegistered(data)
  //       ? alert("check id") 
  //       : checkPassword(data)
  //         ? alert("check password") 
  //         : performLogIn(data);
  // };

  return (
    <>
      <form onSubmit={handleSubmit(onSubmit)}>
        <span>[Login] </span>
        <label>Id: </label>
        <input name="Id" defaultValue="" ref={register} />
        <label>password:</label>
        <input
          name="Password"
          defaultValue=""
          ref={register({ required: true, maxLength: 10 })}
        />
        {errors.exampleRequired && <p>This field is required</p>}
        <input type="submit" />
      </form>
      {loggedIn && <Redirect to={`/${currentUser}/TimeLine`} />}
    </>
  );
};

Login.propTypes = {
  setUserOfActivePage: PropTypes.func,
  setCurrentUser: PropTypes.func,
  toggleLogInStatus: PropTypes.func,
  setLoggedIn: PropTypes.func,
  loggedIn: PropTypes.bool,
  currentUser: PropTypes.string
};

Login.defaultProps = {
  setUserOfActivePage: {},
  setCurrentUser: {},
  toggleLogInStatus: {},
  setLoggedIn: {},
  loggedIn: false,
  currentUser: ""
};

export default Login;
