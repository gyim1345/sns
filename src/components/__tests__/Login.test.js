import React from 'react';
import { mount } from 'enzyme';
import Login from '../Login';
import { currentUser } from '../../Variables';
import { BrowserRouter as Router } from 'react-router-dom';

import MutationObserver from 'mutation-observer'
global.MutationObserver = MutationObserver 

describe('<Login />', () => {
  it('스냅샷 비교', () => {
    const wrapper = mount(
      <Router key="TEST_KEY">
        <Login currentUser={currentUser} loggedIn={true} />
      </Router>
    );

    expect(wrapper.props().children.props.currentUser).toBe(currentUser);
    expect(wrapper.props().children.props.loggedIn).toBe(true);
    expect(wrapper.text()).toMatch('Bongstagram');
    expect(wrapper.text()).toMatch('또는');
    expect(wrapper.html()).toMatch('<form');
    expect(wrapper.html()).toMatchSnapshot();
  });
});
