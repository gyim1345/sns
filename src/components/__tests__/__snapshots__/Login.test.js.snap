// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> 스냅샷 비교 1`] = `"<div class=\\"css-1aghmg0-box-Login\\"><h1 class=\\"css-1tzienn-SNS em34l5a0\\"> Bongstagram </h1><form class=\\"LoginonSubmit css-1muyfbe-box-Login\\"><input name=\\"Id\\" placeholder=\\"전화번호, 사용자 이름, 이메일\\" class=\\"css-15odpl8-inputBoxCss-Login\\" value=\\"\\"><input type=\\"password\\" name=\\"Password\\" placeholder=\\"비밀번호\\" class=\\"css-1ng5cnc-inputBoxCss-Login\\" value=\\"\\"><input type=\\"submit\\" class=\\"css-n6vlve-button-blackButton-Login\\" value=\\"로그인\\"><span class=\\"css-xwj2yi-fontCss-Login\\">또는</span></form></div>"`;
